File: tools/upload-main.js
#!/usr/bin/env node

import path from "path";
import chalk from "chalk";
import { askBlocPath, askTargetDir, askThemeName, askGlobalJsonName } from "./command-line.js";
import { copyBlocFiles } from "./copyFiles.js";
import { readJson, writeJson, addBlocToGlobalJson } from "./jsonUtils.js";
import fs from "fs-extra";



export default async function main() {
    console.log(chalk.blue.bold("\n🚀 Script d’export de bloc ACF\n"));

    // collecte des infos
    const blocPath = await askBlocPath();
    const blocName = path.basename(blocPath);

    if (!await fs.pathExists(blocPath)) {
        console.error(chalk.red(`❌ Dossier source du bloc introuvable → ${blocPath}`));
        process.exit(1);
    }

    const targetDir = await askTargetDir();
    const themeName = await askThemeName();
    const themeBasePath = path.join(targetDir, "wp-content/themes", themeName);

    // vérification du JSON
    const globalJsonName = await askGlobalJsonName();
    const blocJsonPath = path.join(blocPath, "acf", `${blocName}.json`);
    const globalJsonPath = path.join(themeBasePath, "acf", globalJsonName);

    if (!await fs.pathExists(blocJsonPath)) {
        console.error(chalk.red(`❌ JSON du bloc introuvable → ${blocJsonPath}`));
        process.exit(1);
    }

    if (!await fs.pathExists(globalJsonPath)) {
        console.error(chalk.red(`❌ JSON global introuvable → ${globalJsonPath}`));
        process.exit(1);
    }

    const blocJson = await readJson(blocJsonPath);
    const globalJson = await readJson(globalJsonPath);

    console.log(chalk.cyan(`\n Export du bloc "${blocName}" vers le projet ${targetDir} dans le thème ${themeName}\n`));


    // insertion des fichiers
    try {
        // copier les fichiers
        const mappings = [
            { label: "JavaScript", src: path.join(blocPath, "js"), dest: path.join(themeBasePath, "resources/js/Components") },
            { label: "SCSS", src: path.join(blocPath, "scss"), dest: path.join(themeBasePath, "resources/scss/templates/content") },
            { label: "PHP/HTML", src: path.join(blocPath, "php"), dest: path.join(themeBasePath, "templates/template-parts/content") }
        ];
        await copyBlocFiles(mappings);

        // Mettre à jour le JSON du projet
        addBlocToGlobalJson(globalJson, blocJson);
        await writeJson(globalJsonPath, globalJson);

        console.log(chalk.green.bold(`\n ✅ Le bloc "${blocName}" a été ajouté dans ${themeBasePath}`));
    } catch (err) {
        console.error(chalk.red("❌ Erreur pendant l’export, annulation :", err));
        process.exit(1);
    }
}

File: tools/upload_main.py
from pathlib import Path
from .command_line import CommandLineInterface
from .copy_files import FileCopier
from .json_utils import JsonManager
import sys


class UploadMain:

    def __init__(self):
        self.cli = CommandLineInterface()
        self.file_copier = FileCopier()
        self.json_manager = JsonManager()

    def run(self):
        print("\n Script d’export de bloc ACF\n")

        # --- sélection du bloc ---
        bloc_path = Path(self.cli.ask_bloc_path())
        bloc_name = bloc_path.name

        if not bloc_path.exists():
            print(f"❌ Dossier source du bloc introuvable → {bloc_path}")
            sys.exit(1)

        # --- sélection du thème WordPress ---
        theme_name = self.cli.ask_theme_name()
        theme_path = Path(theme_name)
        if not theme_path.exists():
            print(f"❌ Dossier du thème introuvable → {theme_path}")
            sys.exit(1)

        # --- sélection du fichier global JSON ---
        global_json_name = self.cli.ask_global_json_name()
        bloc_json_path = bloc_path / "acf" / f"{bloc_name}.json"
        global_json_path = theme_path / "acf" / global_json_name

        if not bloc_json_path.exists():
            print(f"❌ Fichier JSON du bloc introuvable → {bloc_json_path}")
            sys.exit(1)
        if not global_json_path.exists():
            print(f"❌ Fichier JSON global introuvable → {global_json_path}")
            sys.exit(1)

        bloc_json = self.json_manager.read_json(bloc_json_path)
        global_json = self.json_manager.read_json(global_json_path)

        print(f"\n🚀 Export du bloc \"{bloc_name}\" vers le thème \"{theme_path.name}\"\n")

        try:
            # --- copie des fichiers ---
            # dans UploadMain.run()
            mappings = [
                {"label": "JavaScript", "src": bloc_path / "js", "dest": Path(self.file_copier.custom_paths.get("js"))},
                {"label": "SCSS", "src": bloc_path / "scss", "dest": Path(self.file_copier.custom_paths.get("scss"))},
                {"label": "PHP/HTML", "src": bloc_path / "php", "dest": Path(self.file_copier.custom_paths.get("php"))},
            ]

            self.file_copier.copy_bloc_files(mappings)

            # --- ajout du bloc dans le JSON global ---
            self.json_manager.add_bloc_to_global_json(global_json, bloc_json)
            self.json_manager.write_json(global_json_path, global_json)

            print(f"\n✅ Bloc \"{bloc_name}\" ajouté avec succès dans le thème {theme_path}")
        
        except Exception as e:
            print(f"❌ Erreur pendant l’export : {e}")
            sys.exit(1)
